numpy/_core/src/multiarray/stringdtype/casts.c

```diff
@@ -1394,6 +1406,143 @@ static PyType_Slot dt2s_slots[] = {
+    npy_timedelta *out = (npy_timedelta *)data[1];
+
+    npy_intp in_stride = strides[0];
+    npy_intp out_stride = strides[1] / sizeof(npy_timedelta);
+
+    while (N--) {
+        const npy_packed_static_string *ps = (npy_packed_static_string *)in;
+        npy_static_string s = {0, NULL};
+        int is_null = NpyString_load(allocator, ps, &s);
+        if (is_null == -1) {
+            PyErr_SetString(
+                    PyExc_MemoryError,
+                    "Failed to load string in string to datetime cast");
+            goto fail;
+        }
+        if (is_null || is_nat_string(&s)) {
+            if (has_null && !has_string_na) {
+                *out = NPY_DATETIME_NAT;
+                goto next_step;
+            }
+            s = *default_string;
+        }
+
+        PyObject *pystr = PyUnicode_FromStringAndSize(s.buf, s.size);
+
+        if (pystr == NULL) {
+            goto fail;
+        }
+
+        // interpret as integer in base 10
+        PyObject *pylong_value = PyLong_FromUnicodeObject(pystr, 10);
+
+        Py_DECREF(pystr);
+
+        if (pylong_value == NULL) {
+            goto fail;
+        }
+
+        npy_longlong value = PyLong_AsLongLong(pylong_value);
+
+        Py_DECREF(pylong_value);
+
+        if (value == -1 && PyErr_Occurred()) {
+            goto fail;
+        }
+
+        *out = (npy_timedelta)value;
+
+    next_step:
+        in += in_stride;
+        out += out_stride;
+    }
+
+    NpyString_release_allocator(allocator);
+    return 0;
+
+fail:
+    NpyString_release_allocator(allocator);
+    return -1;
+}
+
+static PyType_Slot s2td_slots[] = {
+        {NPY_METH_resolve_descriptors,
+         &string_to_datetime_timedelta_resolve_descriptors},
+        {NPY_METH_strided_loop, &string_to_timedelta},
+        {0, NULL}};
+
+static char *s2td_name = "cast_StringDType_to_Timedelta";
+
+// timedelta to string
+
+static int
+timedelta_to_string(PyArrayMethod_Context *context, char *const data[],
+                   npy_intp const dimensions[], npy_intp const strides[],
+                   NpyAuxData *NPY_UNUSED(auxdata))
+{
+    npy_intp N = dimensions[0];
+    npy_timedelta *in = (npy_timedelta *)data[0];
+    char *out = data[1];
+
+    npy_intp in_stride = strides[0] / sizeof(npy_timedelta);
+    npy_intp out_stride = strides[1];
+
+    PyArray_StringDTypeObject *sdescr = (PyArray_StringDTypeObject *)context->descriptors[1];
+    npy_string_allocator *allocator = NpyString_acquire_allocator(sdescr);
+
+    while (N--) {
+        npy_packed_static_string *out_pss = (npy_packed_static_string *)out;
+        if (*in == NPY_DATETIME_NAT)
+        {
+            if (NpyString_pack_null(allocator, out_pss) < 0) {
+                npy_gil_error(
+                        PyExc_MemoryError,
+                        "Failed to pack string in datetime to string cast");
+                goto fail;
+            }
+        }
+
+        if (int_to_stringbuf((long long)*in, out, allocator) != 0) {
```

Comment:
this should be an `else if`, no? Otherwise the `NaT` gets interpreted as a integer here.